source/app.d

```diff
@@ -97,49 +100,38 @@ class ImageFitness
     }
 
     import ldc.attributes;
-    @fastmath void rasterize(ref GeneticImage image, in Circle[] circles)
+    @fastmath void rasterizeCircle(ref GeneticImage image, in Circle circle)
     {
-        foreach(ref circle; circles)
+        import std.algorithm : min;
+
+        auto color = circle.color;
+        auto rad2 = circle.radius / 2;
+        auto left = circle.x < rad2 ? 0 : circle.x - rad2;
+        auto right = min(circle.x + rad2 + 1, image.w);
+        auto top = circle.y < rad2 ? 0 : circle.y - rad2;
+        auto bottom = min(circle.y + rad2 + 1, image.h);
+
+        foreach(x; left..right)
         {
-            auto color = circle.color;
-            foreach(x; (circle.x - circle.radius / 2) .. (circle.x + circle.radius / 2))
+            foreach(y; top..bottom)
             {
-                foreach(y; (circle.y - circle.radius / 2) .. (circle.y + circle.radius / 2))
+                if ((x - circle.x) * (x - circle.x) + (y - circle.y) * (y - circle.y) <= rad2 * rad2)
```

Comment:
It would and it actually produces the same code.

The problem with the original `distance` is that calculating a square root is expensive.